src/capabilities.js

```diff
@@ -280,11 +281,23 @@ export class Capabilities {
     if (!ownCapabilities.roleAssignment.includes(roleId)) {
       throw new Error('No capability to assign role ' + roleId)
     }
-    await this.#dataType[kCreateWithDocId](deviceId, {
-      schemaName: 'role',
-      roleId,
-      fromIndex,
-    })
+
+    const existingRoleDoc = await this.#dataType
+      .getByDocId(deviceId)
+      .catch(() => null)
+
+    if (existingRoleDoc) {
+      await this.#dataType.update(existingRoleDoc.versionId, {
+        ...valueOf(existingRoleDoc),
```

Comment:
This should use the current `fromIndex` - the role applies from the `fromIndex` calculated above. Better not use the existingRoleDoc, just use the same value as when creating.

Reply:
addresse via https://github.com/digidem/mapeo-core-next/pull/423/commits/120eea5c700ca5b1b1244e1a6ac316ec72b19aac